src/KafkaW/Consumer.cpp

```diff
@@ -1,198 +1,126 @@
 #include "Consumer.h"
+#include "MetadataException.h"
 #include "logger.h"
-#include <atomic>
-#include <helper.h>
+#include <algorithm>
+#include <iostream>
+#ifdef _MSC_VER
+#include "process.h"
+#define getpid _getpid
+#else
+#include <unistd.h>
+#endif
 
 namespace KafkaW {
-
-static std::atomic<int> g_kafka_consumer_instance_count;
-
-#define KERR(RK, ERR)                                                          \
-  if ((ERR) != 0) {                                                            \
-    LOG(Sev::Error, "Kafka {}  error: {}, {}, {}", rd_kafka_name(RK), ERR,     \
-        rd_kafka_err2name((rd_kafka_resp_err_t)ERR),                           \
-        rd_kafka_err2str((rd_kafka_resp_err_t)ERR));                           \
+Consumer::Consumer(BrokerSettings &BrokerSettings)
+    : ConsumerBrokerSettings(std::move(BrokerSettings)) {
+  std::string ErrorString;
+
+  Conf = RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL);
```

Comment:
```suggestion
  Conf = std::unique_ptr<RdKafka::Conf>(RdKafka::Conf::create(RdKafka::Conf::CONF_GLOBAL));
```
Should not be raw pointer.